neurolang/probabilistic/dalvi_suciu_lift.py

```diff
@@ -135,6 +142,12 @@ def solve_succ_query(query, cpl_program):
                 "Query %s not liftable, algorithm can't be applied",
                 query
             )
+        # project on query's head variables
+        ra_query = _project_on_query_head(ra_query, shattered_query)
+        # re-introduce head variables potentially removed by unification
+        ra_query = _maybe_reintroduce_head_variables(
+            ra_query, flat_query, unified_query
+        )
```

Comment:
This is done to make sure the answered tuples match the head of the rule (similarly to what we did in the small dichotomy theorem). This fixes issues with head of queries containing constants for example